src/main/java/mate/web/AuthenticationFilter.java

```diff
@@ -0,0 +1,42 @@
+package mate.web;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private final Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        HttpSession session = req.getSession();
+        Long userId = (Long) session.getAttribute("user_id");
+        if (userId == null && allowedUrls.contains(req.getServletPath())) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        if (userId == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
```

Comment:
Let's use one **if** statement. We can use userId == null && !allowedUrls.contains(req.getServletPath()) in condition